components/prism-nevod.js

```diff
@@ -0,0 +1,112 @@
+Prism.languages.nevod = {
+	'comment': /\/\/.*|(?:\/\*[\s\S]*?(?:\*\/|$))/,
+	'string': {
+		pattern: /(?:"(?:""|[^"])*"(?!")|'(?:''|[^'])*'(?!'))!?\*?/,
+		greedy: true,
+		inside: {
+			'string-attrs': /!$|!\*$|\*$/,
+		},
+	},
+	'namespace': {
+		pattern: /(@namespace\s+)[a-zA-Z0-9\-.]+(?=\s*{)/,
+		lookbehind: true,
+	},
+	'pattern': {
+		pattern: /(@pattern\s+)?#?[a-zA-Z0-9\-.]+(?:\s*(?:[(]\s*(?:(?:~\s*)?[a-zA-Z0-9\-.]+\s*(?:,\s*(?:(?:~\s*)?[a-zA-Z0-9\-.]*))*)[)]))?(?=\s*=)/,
```

Comment:
1. This pattern contains a few unnecessary non-capturing groups.
2. You were very careful to allowing `\s*` between token but it seems like you forgot a `\s*` after the last `[a-zA-Z0-9\-.]*`. Am I correct to assume that you initially had a `\s*` there but removed it because of the exponential/polynomial backtracking test?

```suggestion
		pattern: /(@pattern\s+)?#?[a-zA-Z0-9\-.]+(?:\s*[(]\s*(?:~\s*)?[a-zA-Z0-9\-.]+\s*(?:,\s*(?:~\s*)?[a-zA-Z0-9\-.]*)*[)])?(?=\s*=)/,
```

Reply:
1. Thank you, will fix.
2. Yes, this was the reason.

Reply:
Fixed.